src/explorer.api/Controllers/ExploreController.cs

```diff
@@ -68,22 +49,43 @@ public class ExploreController : ControllerBase
         public async Task<IActionResult> Result(
             Guid explorationId)
         {
-            Exploration exploration;
-            ExplorationStatus explorationStatus;
+            if (!explorationRegistry.IsRegistered(explorationId))
+            {
+                return NotFound($"Couldn't find exploration with id {explorationId}.");
+            }
 
-            try
+            var explorationStatus = explorationRegistry.GetStatus(explorationId);
+            var explorationParams = explorationRegistry.GetExplorationParams(explorationId);
```

Comment:
You mean for example the exploration could be removed in one thread while another is accessing it? 

I suppose so... 

We also dispose the `Exploration` at the end of this method - I guess this could cause similar issues. We should probably protect the `/result` and `/cancel` endpoints with a semaphore or something like this. 

I don't think this is very likely to happen though since a single exploration is unlikely to be requested multiple times concurrently. Unless I'm missing something?

We can fix this as part of #261

Reply:
Yes, I agree, this is not very likely to happen.